README.md

```diff
@@ -72,13 +72,13 @@ Next, you may either embed the provided matrix with:
 	tapkee::with((method=Isomap,num_neighbors=15)).embedUsing(matrix);
 
 Or provide callbacks (kernel, distance and features) using any combination
-of the `withKernel(KernelCallback)`, `withDistance(DistanceCallback)` and
-`withFeatures(FeaturesCallback)` member functions:
+of the `with(KernelCallback)`, `with(DistanceCallback)` and
+`with(FeaturesCallback)` member functions:
 
 	tapkee::with((method=Isomap,num_neighbors=15))
-	       .withKernel(kernel_callback)
-	       .withDistance(distance_callback)
-	       .withFeatures(features_callback)
+	       .with(kernel_callback)
```

Comment:
The thing that worries me is that both kernel and distance have exactly the same interface and explicit naming might help to avoid mistakes.

Reply:
Their interface is actually different since kernel has kernel(a, b) and distance, distance(a, b). What is an issue is that  their arguments are a template in the dummy callbacks. Also the specialization one.

Reply:
Yeah you're right.